android/src/main/kotlin/com/bbflight/background_downloader/TaskWorker.kt

```diff
@@ -101,14 +102,21 @@ open class TaskWorker(
                         } else {
                             argList.add(arg)
                         }
+                        /**
                         if (BDPlugin.backgroundChannel != null) {
                             BDPlugin.backgroundChannel?.invokeMethod(
                                 method, argList
                             )
                             if (!BDPlugin.forceFailPostOnBackgroundChannel) {
                                 success.complete(true)
                             }
-                        } else {
+                        }
+                        **/
+                         var channel :MethodChannel? = BDPlugin.bgChannelsByTask[task.taskId];
```

Comment:
If we follow the above, then we don't need/have bgChannelsByTask. Instead we need the context, so we'd have to pass that into the `postOnBackgroundChannel` function, which I think can be done